infrastructure/lambdaEdge.ts

```diff
@@ -34,64 +38,80 @@ export class LambdaEdge extends pulumi.ComponentResource {
-                            "edgelambda.amazonaws.com",
-                        ],
-                    },
-                }],
-                Version: "2012-10-17",
+        this.role = new aws.iam.Role(`${this.name}-lambdaEdgeRole`,
+            {
+                assumeRolePolicy: {
+                    Statement: [{
+                        Effect: "Allow",
+                        Action: "sts:AssumeRole",
+                        Principal: {
+                            Service: [
+                                "lambda.amazonaws.com",
+                                "edgelambda.amazonaws.com",
+                            ],
+                        },
+                    }],
+                    Version: "2012-10-17",
+                },
+            },
+            {
+                parent: this,
             },
-        // tslint:disable-next-line:align
-        }, { parent: this });
+        );
 
-        const rolePolicy = new aws.iam.RolePolicy("lambdaCloudWatchPolicy", {
-            role: this.role,
-            policy: {
-                Version: "2012-10-17",
-                Statement: [
-                    {
-                    Action: [
-                        "logs:CreateLogStream",
-                        "logs:PutLogEvents",
-                        "logs:CreateLogGroup",
+        const rolePolicy = new aws.iam.RolePolicy(`${this.name}-lambdaCloudWatchPolicy`,
+            {
+                role: this.role,
+                policy: {
+                    Version: "2012-10-17",
+                    Statement: [
+                        {
+                        Action: [
+                            "logs:CreateLogStream",
+                            "logs:PutLogEvents",
+                            "logs:CreateLogGroup",
+                        ],
+                        Effect: "Allow",
+                        Resource: "*",
+                        },
                     ],
-                    Effect: "Allow",
-                    Resource: "*",
-                    },
-                ],
+                },
+            },
+            {
+                parent: this,
             },
-        // tslint:disable-next-line:align
-        }, { parent: this });
+        );
     }
 
     private createLambdaFunction() {
-        this.lambdaEdgeFunc = new aws.lambda.CallbackFunction("lambdaEdge", {
-            callback: this.args.func,
-            description: this.args.funcDescription,
-            // Minimum is 128MB.
-            memorySize: 128,
-            publish: true,
-            role: this.role,
-            runtime: "nodejs12.x",
-            // Note that Lambda@Edge functions have a different max timeout of 30 seconds
-            // than the regular Lambda functions.
-            timeout: 5,
-        // tslint:disable-next-line:align
-        }, { parent: this });
+        this.lambdaEdgeFunc = new aws.lambda.CallbackFunction(`${this.name}-lambdaEdge`,
```

Comment:
> This will most likely cause a problem during deployment.

Yep, I saw this during development: https://pulumi.slack.com/archives/C9SGX9QA1/p1612458433035600

Yeah, that's a bit of a bummer, but the workaround seems reasonable enough. Lemme see what I can do.

Reply:
Seems good, yeah?

<img width="591" alt="image" src="https://user-images.githubusercontent.com/274700/107080487-349f0180-67a6-11eb-8d69-07980c45f32b.png">